pkg/resources/secret_with_oauth_authorization_code_grant_acceptance_test.go

```diff
@@ -60,117 +66,87 @@ func TestAcc_SecretWithAuthorizationCodeGrant_BasicFlow(t *testing.T) {
-				),
-			},
-			// set comment externally
-			{
-				PreConfig: func() {
-					acc.TestClient().Secret.Alter(t, sdk.NewAlterSecretRequest(id).WithSet(*sdk.NewSecretSetRequest().
-						WithComment("secret resource - changed comment"),
-					))
-				},
-				Config: config.FromModel(t, secretModel),
+				Config: config.FromModel(t, secretModelAllSet),
 				ConfigPlanChecks: resource.ConfigPlanChecks{
 					PreApply: []plancheck.PlanCheck{
-						plancheck.ExpectResourceAction(secretModel.ResourceReference(), plancheck.ResourceActionUpdate),
-						planchecks.ExpectChange(secretModel.ResourceReference(), "comment", tfjson.ActionUpdate, sdk.String("secret resource - changed comment"), sdk.String(newComment)),
+						plancheck.ExpectResourceAction(secretName, plancheck.ResourceActionUpdate),
+						planchecks.ExpectChange(secretName, "comment", tfjson.ActionUpdate, nil, sdk.String(comment)),
+						planchecks.ExpectChange(secretName, "oauth_refresh_token", tfjson.ActionUpdate, sdk.String(refreshToken), sdk.String(newRefreshToken)),
+						planchecks.ExpectChange(secretName, "oauth_refresh_token_expiry_time", tfjson.ActionUpdate, sdk.String(refreshTokenExpiryDateTime), sdk.String(newRefreshTokenExpiryDateOnly)),
 					},
 				},
 				Check: resource.ComposeTestCheckFunc(
 					assert.AssertThat(t,
-						resourceassert.SecretWithAuthorizationCodeResource(t, secretModel.ResourceReference()).
+						resourceassert.SecretWithAuthorizationCodeResource(t, secretName).
 							HasNameString(name).
 							HasDatabaseString(id.DatabaseName()).
 							HasSchemaString(id.SchemaName()).
 							HasApiAuthenticationString(integrationId.Name()).
 							HasOauthRefreshTokenString(newRefreshToken).
 							HasOauthRefreshTokenExpiryTimeString(newRefreshTokenExpiryDateOnly).
-							HasCommentString(newComment),
-						assert.Check(resource.TestCheckResourceAttrSet(secretModel.ResourceReference(), "describe_output.0.oauth_refresh_token_expiry_time")),
+							HasCommentString(comment),
+						assert.Check(resource.TestCheckResourceAttrSet(secretName, "describe_output.0.oauth_refresh_token_expiry_time")),
 					),
 				),
 			},
-			// set refresh_token_expiry_time externally
+			// set comment and refresh_token_expiry_time externally
 			{
 				PreConfig: func() {
 					acc.TestClient().Secret.Alter(t, sdk.NewAlterSecretRequest(id).WithSet(*sdk.NewSecretSetRequest().
-						WithSetForFlow(*sdk.NewSetForFlowRequest().WithSetForOAuthAuthorization(*sdk.NewSetForOAuthAuthorizationRequest().WithOauthRefreshTokenExpiryTime(time.Now().Add(24 * time.Hour).Format(time.DateTime)))),
+						WithComment("secret resource - changed comment").
+						WithSetForFlow(*sdk.NewSetForFlowRequest().WithSetForOAuthAuthorization(*sdk.NewSetForOAuthAuthorizationRequest().WithOauthRefreshTokenExpiryTime(time.Now().Add(24 * time.Hour).Format(time.DateOnly)))),
 					))
 				},
-				Config: config.FromModel(t, secretModel),
+				Config: config.FromModel(t, secretModelExternalChanges),
```

Comment:
I am not quite sure about this test, why we alter both the config and externally?

Reply:
because i didnt correct this one after changes to handling `token_expiry_time` field
I will correct that and use properly, it should test the external change